scripts/bert/create_pretraining_data.py

```diff
@@ -412,15 +413,28 @@ def create_instances_from_document(x):
                                           ['index', 'label'])
 
 
-def create_masked_lm_predictions(tokens, masked_lm_prob,
-                                 max_predictions_per_seq, vocab_size,
+def create_masked_lm_predictions(tokens, masked_lm_prob, max_predictions_per_seq,
+                                 do_whole_word_mask, vocab,
                                  _MASK_TOKEN, _CLS_TOKEN, _SEP_TOKEN):
     """Creates the predictions for the masked LM objective."""
     cand_indexes = []
     for (i, token) in enumerate(tokens):
         if token in [_CLS_TOKEN, _SEP_TOKEN]:
             continue
-        cand_indexes.append(i)
+        # Whole Word Masking means that if we mask all of the subwords
+        # corresponding to an original word. When a word has been split into
+        # subwords, the first token does not have any marker and any subsequence
+        # tokens are prefixed with ##. So whenever we see the ## token, we
+        # append it to the previous set of word indexes.
+        #
+        # Note that Whole Word Masking does *not* change the training code
+        # at all -- we still predict each subword independently, softmaxed
+        # over the entire vocabulary.
+        if (do_whole_word_mask and len(cand_indexes) >= 1
+                and vocab.idx_to_token[token].startswith('##')):
```

Comment:
The script also supports two tokenizers:
- BERTTokenizer: https://github.com/dmlc/gluon-nlp/blob/master/src/gluonnlp/data/transforms.py#L966, which uses ## to denote subsequent subwords.
- BERTSPTokenizer: https://github.com/dmlc/gluon-nlp/blob/master/src/gluonnlp/data/transforms.py#L1074, which uses sentencepiece for subword tokenization (see an example in http://gluon-nlp.mxnet.io/master/api/modules/data.html?highlight=sentencepiece#gluonnlp.data.SentencepieceTokenizer), and any word not starting with `▁` is a subword . 

Would you mind adding a `is_segment_start` method to both `BERTTokenizer` and `BERTSPTokenizer` so that we can determine whether the token is the starting of a word programmatically?